iorgen/parser_ocaml.py

```diff
@@ -187,15 +186,45 @@ def method(input_data: Input, reprint: bool) -> List[str]:
 def gen_ocaml(input_data: Input, reprint: bool = False) -> str:
     """Generate a OCaml code to parse input"""
     output = ""
+    main = ""
     for record in input_data.structs:
-        output += "\n".join(declare_record(record))
-        output += "\n\n"
-    output += "\n".join(method(input_data, reprint))
-    output += "\n\nlet () =\n"
+        main += "\n".join(declare_record(record))
+        main += "\n\n"
+    main += "\n".join(method(input_data, reprint))
+    main += "\n\nlet () =\n"
     for var in input_data.input:
-        output += INDENTATION + "let {} = {} in\n".format(
+        main += INDENTATION + "let {} = {} in\n".format(
             var_name(var.name), read_lines(var.type, input_data))
     args = (var_name(i.name) for i in input_data.input)
-    output += "{}{} {}\n".format(INDENTATION, var_name(input_data.name),
-                                 " ".join(args))
+    main += "{}{} {}\n".format(INDENTATION, var_name(input_data.name),
+                               " ".join(args))
+    if "List.init " in main:
+        # This is a quick fix to avoid dependency of OCaml 4.06 for List.init
+        output += "module List = struct\n"
```

Comment:
You choose to put this on the very top of the generated file. I guess it is generally a good practice to do so. But here, if possible, I think it is better to keep it below the function to complete. This is just some boilerplate, and should not be the very first think to be read.

Reply:
There is already type definition at the top. With type at the top and this module hack at the bottom it become hard to find the function to complete but if you think it is better to do so I can move this bellow the function.

Reply:
OK